src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java

```diff
@@ -724,6 +805,332 @@ public final void markDirty()
+    {
+        if(stack != null)
+        {
+            itemsCurrentlyNeeded.add(stack);
+        }
+    }
+
+    /**
+     * Getter for the neededItems.
+     * @return an unmodifiable list.
+     */
+    public List<ItemStack> getNeededItems()
+    {
+        return Collections.unmodifiableList(itemsCurrentlyNeeded);
+    }
+
+    /**
+     * Getter for the first of the currentlyNeededItems.
+     * @return copy of the itemStack.
+     */
+    @Nullable
+    public ItemStack getFirstNeededItem()
+    {
+        if(itemsCurrentlyNeeded.isEmpty())
+        {
+            return null;
+        }
+        return itemsCurrentlyNeeded.get(0).copy();
+    }
+
+    /**
+     * Clear the currentlyNeededItem list.
+     */
+    public void clearNeededItems()
+    {
+        itemsCurrentlyNeeded.clear();
+    }
+
+    /**
+     * Overwrite the itemsCurrentlyNeededList with a new one.
+     * @param newList the new list to set.
+     */
+    public void setItemsCurrentlyNeeded(@NotNull List<ItemStack> newList)
+    {
+        this.itemsCurrentlyNeeded = new ArrayList<>(newList);
+    }
+
+    /**
+     * Set the needed pickaxe level of the worker.
+     * @param needsPickaxeLevel the mining level.
+     */
+    public void setNeedsPickaxeLevel(final int needsPickaxeLevel)
+    {
+        this.needsPickaxeLevel = needsPickaxeLevel;
+    }
+
+    /**
+     * Check for the required tool and return the describing string.
+     * @return the string of the required tool.
+     */
+    public String getRequiredTool()
+    {
+        if(needsHoe)
+        {
+            return Utils.HOE;
+        }
+
+        if(needsAxe)
+        {
+            return Utils.AXE;
+        }
+
+        if(needsPickaxe)
+        {
+            return Utils.PICKAXE;
+        }
+
+        if(needsShovel)
+        {
+            return Utils.SHOVEL;
+        }
+
+        if(needsWeapon)
+        {
+            return Utils.WEAPON;
+        }
+
+        return "";
+    }
+
+    /**
+     * Try to transfer a stack to one of the inventories of the building.
+     * @param stack the stack to transfer.
+     * @param world the world to do it in.
+     * @return true if was able to.
+     */
+    public boolean transferStack(@NotNull final ItemStack stack, @NotNull final World world)
+    {
+        if(tileEntity == null || InventoryUtils.isInventoryFull(tileEntity))
+        {
+            for(final BlockPos pos: containerList)
+            {
+                final TileEntity tempTileEntity = world.getTileEntity(pos);
+                if(tempTileEntity instanceof TileEntityChest && !InventoryUtils.isInventoryFull((IInventory) tempTileEntity))
+                {
+                    return InventoryUtils.addItemStackToInventory((IInventory) tempTileEntity, stack);
+                }
+            }
+        }
+        else
+        {
+            return InventoryUtils.addItemStackToInventory(tileEntity, stack);
+        }
+        return false;
+    }
+
+    /**
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) First sentence should end with a period. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)